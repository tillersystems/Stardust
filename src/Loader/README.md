# Loader

### Usage

```jsx
import { Loader } from '@tillersystems/stardust';
```

<!-- STORY -->

### Properties

- `className` - className needed by styled component.
- `color` - color.
- `height` - Loader height.
- `width` - Loader width.

| propName    | propType | defaultValue | isRequired |
| ----------- | :------: | :----------: | :--------: |
| `className` | `string` |     `''`     |     -      |
| `color`     | `string` |    `blue`    |     -      |
| `height`    | `string` |    `2rem`    |     -      |
| `width`     | `string` |    `2rem`    |     -      |

### Example

```jsx
import { Loader } from '@tillersystems/stardust';

render() {
  return <Loader width="2rem" height="2rem" />
}
```
