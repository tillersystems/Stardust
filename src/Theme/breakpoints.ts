/**
 * Defines application breakpoints.
 */

const xs = 0;
const sm = 576;
const md = 768;
const lg = 992;
const xl = 1200;

export const breakpoints = {
  xs,
  sm,
  md,
  lg,
  xl,
};
