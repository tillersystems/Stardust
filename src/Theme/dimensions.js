/**
 * Defines the theming dimensions.
 */

const radius = 0.4;
const tiny = 0.3;
const small = 0.7;
const medium = 1.5;
const big = 3.1;
const huge = 4.5;

export const dimensions = {
  radius: `${radius}rem`,

  tiny: `${tiny}rem`,
  small: `${small}rem`,
  medium: `${medium}rem`,
  big: `${big}rem`,
  huge: `${huge}rem`,

  radiusInt: radius,

  tinyInt: tiny,
  smallInt: small,
  mediumInt: medium,
  bigInt: big,
  hugeInt: huge,
};
