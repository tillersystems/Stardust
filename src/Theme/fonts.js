/**
 * Defines the theming of fonts.
 */

export const fonts = {
  size: {
    // -start- Old font size
    tiny: '0.3rem',
    small: '0.7rem',
    // medium: '1.2rem',
    // big: '3.2rem',
    huge: '6.4rem',
    // -end- Old font size

    // -start- New font size
    default: '1.2rem',
    medium: '1.4rem',
    big: '1.6rem',
    h6: '1.8rem',
    h5: '2.0rem',
    h4: '2.2rem',
    h3: '2.6rem',
    h2: '3rem',
    h1: '3.6rem',
    // -end- New font size

    // -start- Old font int
    tinyInt: 0.3,
    smallInt: 0.7,
    // mediumInt: 1.2,
    // bigInt: 3.2,
    hugeInt: 6.4,
    // -end- Old font int

    // -start- Old font int
    defaultInt: 1.2,
    mediumInt: 1.4,
    bigInt: 1.6,
    h6Int: 1.8,
    h5Int: 2.0,
    h4Int: 2.2,
    h3Int: 2.6,
    h2Int: 3,
    h1Int: 3.6,
    // -end- Old font int
  },

  spacing: {
    small: '0.5px',
    medium: '1px',
    big: '1.5px',
  },

  weight: {
    thin: 200,
    normal: 400,
    thick: 600,
  },

  family: '"Open Sans", Helvetica, Arial, sans-serif',
};
