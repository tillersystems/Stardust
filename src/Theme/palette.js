/**
 * Defines the theming colors.
 */

import { lighten, darken } from 'polished';

const dark = color => darken(0.03, color);

const light = color => lighten(0.1, color);

// -start- Old colours palette
const white = 'hsl(0, 100%, 100%)';
const gray = 'hsl(218, 18%, 91%)';
const black = 'hsl(0, 0%, 0%)';
const blackLight = 'hsl(213, 17%, 20%)';

// const red = 'hsl(6, 79%, 65%)';
// const green = 'hsl(89, 53%, 52%)';
// const blue = 'hsl(200, 74%, 46%)';
const orange = 'hsl(34, 91%, 65%)';
const anthracite = 'hsl(213, 17%, 20%)';
const clay = 'hsl(215, 17%, 26%)';
const sand = 'hsl(214, 8%, 84%)';
const marble = 'hsl(206, 23%, 69%)';
const grey = 'hsl(206, 16%, 60%)';
const bodyBackground = 'hsl(218, 29%, 95%)';
const emerald = 'hsl(145, 63%, 49%)';
const river = 'hsl(204, 70%, 53%)';
const wisteria = 'hsl(282, 44%, 47%)';
const carrot = 'hsl(28, 80%, 52%)';
const alizarin = 'hsl(6, 78%, 57%)';
const turquoise = 'hsl(168, 76%, 42%)';
const greygoose = 'hsl(216, 29%, 97%)';
const fadeBlue = 'hsl(210, 50%, 98%)';

const forePrimary = white;
// const backPrimary = blue;

const foreSecondary = grey;
const backSecondary = white;

const foreSuccess = white;
// const backSuccess = green;

const foreFailure = white;
// const backFailure = red;

const googleBrandRed = '#ea4335';
// -end- Old colours palette

// -start- New colours palette
const whiteOpacity = opacity => `hsla(0, 100%, 100%, ${opacity})`;

const red = 'hsl(6, 79%, 65%)';
const red1 = 'hsl(6, 60%, 58%)';
const red2 = 'hsl(6, 52%, 55%)';
const green = 'hsl(89, 53%, 52%)';
const green1 = 'hsl(89, 49%, 45%)';
const green2 = 'hsl(89, 49%, 42%)';
const blue = 'hsl(200, 74%, 46%)';
const blue1 = 'hsl(200, 74%, 41%)';
const blue2 = 'hsl(200, 74%, 38%)';
const darkBlue = 'hsl(213, 17%, 20%)';
const yellow = 'hsl(9, 91%, 65%)';

const mysticGrey = 'hsl(216,45%,98%)';
const lightGrey = 'hsl(207,22%,90%)';
const mediumGrey = 'hsl(206,23%,69%)';
const darkGrey = 'hsl(206,16%,60%)';
const spaceGrey = 'hsl(207,13%,45%)';

const backPrimary = blue;
const backFailure = red;
const backSuccess = green;
// -end- New colours palette

export const palette = {
  // -start- Old colours palette
  white,
  black,
  blackLight,

  googleBrandRed,

  gray,
  greygoose,
  darkGray: dark(gray),
  lightGray: light(gray),

  red,
  darkRed: dark(red),
  lightRed: light(red),

  green,
  darkGreen: dark(green),
  lightGreen: light(green),

  blue,
  // darkBlue: dark(blue),
  lightBlue: light(blue),

  orange,

  anthracite,
  clay,
  sand,
  marble,
  emerald,
  river,
  wisteria,
  carrot,
  alizarin,
  turquoise,
  grey,
  bodyBackground,
  fadeBlue,

  primary: {
    fore: forePrimary,
    darkFore: dark(forePrimary),
    lightFore: light(forePrimary),

    back: backPrimary,
    darkBack: dark(backPrimary),
    lightBack: light(backPrimary),

    default: blue,
    dark: blue1,
    darker: blue2,
  },

  secondary: {
    fore: foreSecondary,
    darkFore: dark(foreSecondary),
    lightFore: light(foreSecondary),

    back: backSecondary,
    darkBack: dark(backSecondary),
    lightBack: light(backSecondary),
  },

  success: {
    fore: foreSuccess,
    darkFore: dark(foreSuccess),
    lightFore: light(foreSuccess),

    back: backSuccess,
    darkBack: dark(backSuccess),
    lightBack: light(backSuccess),

    default: green,
    dark: green1,
    darker: green2,
  },

  failure: {
    fore: foreFailure,
    darkFore: dark(foreFailure),
    lightFore: light(foreFailure),

    back: backFailure,
    darkBack: dark(backFailure),
    lightBack: light(backFailure),

    default: red,
    dark: red1,
    darker: red2,
  },
  // -end- Old colours palette

  // -start- New colours palette
  whiteOpacity,
  // red,
  // green,
  // blue,
  darkBlue,
  yellow,

  mysticGrey,
  lightGrey,
  mediumGrey,
  darkGrey,
  spaceGrey,
  // -end- New colours palette
};
