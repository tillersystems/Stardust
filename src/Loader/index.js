import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

/**
 * Loader
 *
 * This component is in charge of displaying
 * an Loader to indicate that an action is in progress
 *
 * @param {string} className // Classes needed by styled component.
 * @param {string} color // The color of the Loader.
 * @param {string} width // The width of the Loader.
 * @param {string} height // The height of the Loader.
 *
 * @return {jsx}
 */

import Theme from '../Theme';

const Loader = ({ className, color, width, height }) => (
  <svg className={className} viewBox="0 0 50 50" width={width} height={height} aria-hidden="true">
    <circle
      className="path"
      cx="25"
      cy="25"
      r="20"
      fill="none"
      stroke={color}
      strokeWidth="4"
      data-testid="circleShape"
    />
  </svg>
);

/**
 * PropTypes Validation
 */
const { string } = PropTypes;
Loader.propTypes = {
  className: string,
  color: string,
  height: string,
  width: string,
};

/**
 * Default Props
 */
Loader.defaultProps = {
  className: '',
  color: Theme.palette.primary.default,
  height: '20px',
  width: '20px',
};

export default styled(Loader)`
  animation: rotate 1.4s linear infinite;
  display: block;

  & .path {
    stroke-linecap: round;
    animation: dash 1.4s ease-in-out infinite;
  }

  @keyframes rotate {
    100% {
      transform: rotate(360deg);
    }
  }

  @keyframes dash {
    0% {
      stroke-dasharray: 1, 150;
      stroke-dashoffset: 0;
    }
    50% {
      stroke-dasharray: 90, 150;
      stroke-dashoffset: -35;
    }
    100% {
      stroke-dasharray: 90, 150;
      stroke-dashoffset: -124;
    }
  }
`;
