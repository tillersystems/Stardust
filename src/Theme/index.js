/**
 * Theme exported to the provider
 * This is globally available in styled-components when interpolating a function like so:
 * ${(props) => props.theme}
 * Or using import { withTheme } from 'styled-components';
 */

import { breakpoints } from './breakpoints';
import { palette } from './palette';
import { dimensions } from './dimensions';
import { fonts } from './fonts';

const Theme = {
  breakpoints,
  palette,
  dimensions,
  fonts,
};

export default Theme;
