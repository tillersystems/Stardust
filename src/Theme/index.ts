import { DefaultTheme } from 'styled-components';

/**
 * Theme exported to the provider
 * This is globally available in styled-components when interpolating a function like so:
 * ${(props) => props.theme}
 * Or using import { withTheme } from 'styled-components';
 */

import { breakpoints } from './breakpoints';
import { palette } from './palette';
import { dimensions } from './dimensions';
import { fonts } from './fonts';

const Theme: DefaultTheme = {
  breakpoints,
  palette,
  dimensions,
  fonts,
};

export default Theme;
