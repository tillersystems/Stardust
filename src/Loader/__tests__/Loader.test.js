import 'jest-styled-components';
import React from 'react';

import Loader from '..';
import Theme from '../../Theme';

describe('<Loader />', () => {
  it('should render without a problem ', () => {
    const render = shallow(<Loader />);

    expect(render.dive()).toMatchSnapshot();
  });

  it('should render with another size ', () => {
    const render = shallow(<Loader width="4rem" height="4rem" />);

    expect(render.dive()).toMatchSnapshot();
  });

  it('should render with another color ', () => {
    const render = shallow(<Loader color={Theme.palette.failure.default} />);

    expect(render.dive()).toMatchSnapshot();
  });
});
