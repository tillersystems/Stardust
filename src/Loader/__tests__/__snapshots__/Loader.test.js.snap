// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> should render with another color  1`] = `
.c0 {
  -webkit-animation: rotate 1.4s linear infinite;
  animation: rotate 1.4s linear infinite;
  display: block;
}

.c0 .path {
  stroke-linecap: round;
  -webkit-animation: dash 1.4s ease-in-out infinite;
  animation: dash 1.4s ease-in-out infinite;
}

<svg
  aria-hidden="true"
  className="c0"
  height="2rem"
  viewBox="0 0 50 50"
  width="2rem"
>
  <circle
    className="path"
    cx="25"
    cy="25"
    fill="none"
    r="20"
    stroke="hsl(6, 79%, 65%)"
    strokeWidth="4"
  />
</svg>
`;

exports[`<Loader /> should render with another size  1`] = `
.c0 {
  -webkit-animation: rotate 1.4s linear infinite;
  animation: rotate 1.4s linear infinite;
  display: block;
}

.c0 .path {
  stroke-linecap: round;
  -webkit-animation: dash 1.4s ease-in-out infinite;
  animation: dash 1.4s ease-in-out infinite;
}

<svg
  aria-hidden="true"
  className="c0"
  height="4rem"
  viewBox="0 0 50 50"
  width="4rem"
>
  <circle
    className="path"
    cx="25"
    cy="25"
    fill="none"
    r="20"
    stroke="hsl(200, 74%, 46%)"
    strokeWidth="4"
  />
</svg>
`;

exports[`<Loader /> should render without a problem  1`] = `
.c0 {
  -webkit-animation: rotate 1.4s linear infinite;
  animation: rotate 1.4s linear infinite;
  display: block;
}

.c0 .path {
  stroke-linecap: round;
  -webkit-animation: dash 1.4s ease-in-out infinite;
  animation: dash 1.4s ease-in-out infinite;
}

<svg
  aria-hidden="true"
  className="c0"
  height="2rem"
  viewBox="0 0 50 50"
  width="2rem"
>
  <circle
    className="path"
    cx="25"
    cy="25"
    fill="none"
    r="20"
    stroke="hsl(200, 74%, 46%)"
    strokeWidth="4"
  />
</svg>
`;
